src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +63,990 @@ const ActorInit Dm_Char08_InitVars = {
+        this->dynapolyInitialized = true;
+        if (this) {}
+    }
+
+    this->palmTree1 = Actor_Spawn(&play->actorCtx, play, ACTOR_OBJ_YASI, this->dyna.actor.world.pos.x + -80.0f,
+                                  this->dyna.actor.world.pos.y + 390.0f, this->dyna.actor.world.pos.z, 0, 0, 0, 1);
+    this->palmTree2 =
+        Actor_Spawn(&play->actorCtx, play, ACTOR_OBJ_YASI, this->dyna.actor.world.pos.x + 68.0f,
+                    this->dyna.actor.world.pos.y + 368.0f, this->dyna.actor.world.pos.z - 174.0f, 0, 0x7530, 0, 1);
+    switch (play->sceneNum) {
+        case SCENE_31MISAKI:
+            if (gSaveContext.save.weekEventReg[53] & 0x20) {
+                this->dyna.actor.world.pos.x = -6480.0f;
+                this->targetYPos = -120.0f;
+                this->dyna.actor.world.pos.z = 1750.0f;
+                this->dyna.actor.world.rot.x = 0;
+                this->dyna.actor.world.rot.y = -0x7234;
+                this->dyna.actor.world.rot.z = 0;
+                this->dyna.actor.shape.rot.x = 0;
+                this->dyna.actor.shape.rot.y = -0x7234;
+                this->dyna.actor.shape.rot.z = 0;
+                this->unk_1F0 = 1.0f;
+                this->unk_1FF = 2;
+                this->animIndex = ANIM_2;
+                this->unk_203 = 0x63;
+                this->eyeMode = EYEMODE_1;
+                this->unk_207 = 0;
+                this->unk_208 = 0;
+                this->dyna.actor.flags |= ACTOR_FLAG_1;
+                if (gSaveContext.save.entranceIndex == 0x6A80) {
+                    this->eyeMode = EYEMODE_0;
+                    this->actionFunc = func_80AAFAC4;
+                } else {
+                    this->actionFunc = func_80AAF8F4;
+                }
+            } else {
+                this->actionFunc = DmChar08_WaitForSong;
+            }
+            break;
+        case SCENE_SEA:
+            this->unk_1FF = 2;
+            this->animIndex = ANIM_2;
+            this->unk_203 = 0x63;
+            this->eyeMode = EYEMODE_0;
+            this->unk_207 = 0;
+            this->unk_208 = 0;
+            this->dyna.actor.flags |= ACTOR_FLAG_1;
+            this->actionFunc = func_80AAFAE4;
+            this->unk_1F0 = 1.0f;
+            break;
+        case SCENE_KONPEKI_ENT:
+            this->unk_1FF = 2;
+            this->animIndex = ANIM_2;
+            this->unk_203 = 0x63;
+            this->eyeMode = EYEMODE_0;
+            this->unk_207 = 0;
+            this->unk_208 = 0;
+            this->actionFunc = func_80AAFE78;
+            this->unk_1F0 = 1.0f;
+            break;
+    }
+    DmChar08_SetAnimation(&this->skelAnime, &sLargeTurtleAnimationInfo[this->animIndex], 0);
+}
+
+void DmChar08_Destroy(Actor* thisx, PlayState* play) {
+    DmChar08* this = THIS;
+
+    if (this->dynapolyInitialized) {
+        DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+    }
+}
+
+s16 sBigTurtleOcarinaSoundLatch = false;
```

Comment:
What does this mean?

Reply:
from what I can tell it makes it so that when you pull the ocarina the action func running constantly doesn't cause the ocarina sound to play constantly.